lib/customizer.php

```diff
@@ -55,10 +55,12 @@ function gutenberg_customize_register( $wp_customize ) {
 			'sanitize_callback' => 'gutenberg_customize_sanitize',
 		)
 	);
-	$wp_customize->add_section(
-		'gutenberg_widget_blocks',
-		array( 'title' => __( 'Widget Blocks (Experimental)', 'gutenberg' ) )
-	);
+	if ( get_option( 'gutenberg-experiments' ) && array_key_exists( 'gutenberg-widget-experiments', get_option( 'gutenberg-experiments' ) ) ) {
+		$wp_customize->add_section(
+			'gutenberg_widget_blocks',
+			array( 'title' => __( 'Widget Blocks (Experimental)', 'gutenberg' ) )
+		);
+	}
 	$wp_customize->add_control(
```

Comment:
Shouldn't we bail out of adding anything related to the widget blocks?

Adding the controls and filters without the section might still cause issues.

Reply:
Good point, I'll move it inside the condition.

Reply:
It's not just the section and controls though.

There are a lot of filters and actions there that modify legacy widget logic. Are we sure we are not breaking any expectations by leaving them there. Can we not just *not load any file related to the customizer in `load.php`*?

cc @jorgefilipecosta

Reply:
Hmm, that doesn't work. Loading `customizer.php` conditionally causes the whole customizer page to crash when the widget experiment is disabled. The absence of `add_filter( 'widget_customizer_setting_args', 'filter_widget_customizer_setting_args' );` is what does it, though I have no idea why. Not loading `widgets.php` also causes problems when the experiment is disabled: the page loads with a warning that `$sidebar_widget_ids` in `class-wp-customize-widgets.php` is the wrong type. That one seems to be caused by the absence of `add_filter( 'sidebars_widgets', 'Experimental_WP_Widget_Blocks_Manager::swap_out_sidebars_blocks_for_block_widgets' );` 🤷‍♀ 
In any case, it's better to load `customizer.php` by default because even though it's only being used for widget-related stuff now, that might change in the future, so isolating the widget logic within it now will save us work later.
I'd still like to know why not loading these filters causes everything to explode though 😁 . Maybe @jorgefilipecosta can shed some light on that?

Reply:
It's because the filters work together to change widget areas from storing an array of widgets to store a post ID. That's why I was thinking we should disable loading the entire files. I'm not sure if leaving some of the filters active could cause some weird side effects.

I guess that if we test it well enough and @jorgefilipecosta doesn't see any problems from his side then I'm happy to merge this.